src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: object[],
+  key: keyof object,
+): GroupsMap<object> {
```

Comment:
to use generic you need to point what function will receive and then use it in arguments
you can try like this http://joxi.ru/4AkdqYQT0Zb3Dr